applications/asset_tracker_v2/tests/lwm2m_codec/src/lwm2m_codec_test.c

```diff
@@ -0,0 +1,769 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <unity.h>
+#include <stdbool.h>
+#include <string.h>
+#include <errno.h>
+#include <net/lwm2m.h>
+#include <net/lwm2m_client_utils.h>
+
+#include "lwm2m_client_utils/mock_lwm2m_client_utils.h"
+#include "lwm2m/mock_lwm2m.h"
+#include "lwm2m/mock_lwm2m_resource_ids.h"
+#include "lte_lc/mock_lte_lc.h"
+#include "date_time/mock_date_time.h"
+
+#include "cloud_codec.h"
+#include "lwm2m_codec_defines.h"
+
+/* Default test configuration values. */
+#define GNSS_TIMEOUT 30
+#define ACTIVE_MODE true
+#define ACTIVE_WAIT_TIMEOUT 120
+#define MOVEMENT_RESOLUTION 120
+#define MOVEMENT_TIMEOUT 3600
+#define ACCELEROMETER_THRES 10
+#define NOD_NEIGHBOR_CELL false
+#define NOD_GNSS true
+
+static uint8_t bearers[2] = { LTE_FDD_BEARER, NB_IOT_BEARER };
+static int battery_voltage;
+static int32_t pressure_ts;
+static int32_t temperature_ts;
+static int32_t humidity_ts;
+static int32_t button_ts;
+static int64_t current_time;
+static double temp_min_range_val = BME680_TEMP_MIN_RANGE_VALUE;
+static double temp_max_range_val = BME680_TEMP_MAX_RANGE_VALUE;
+static double humid_min_range_val = BME680_HUMID_MIN_RANGE_VALUE;
+static double humid_max_range_val = BME680_HUMID_MAX_RANGE_VALUE;
+static double pressure_min_range_val = BME680_PRESSURE_MIN_RANGE_VALUE;
+static double pressure_max_range_val = BME680_PRESSURE_MAX_RANGE_VALUE;
+
+/* Callback handler used to trigger configuration updates in the uut. */
+static lwm2m_engine_set_data_cb_t post_write_cb;
+
+extern int unity_main(void);
+extern int generic_suiteTearDown(int num_failures);
+
+/* Suite teardown shall finalize with mandatory call to generic_suiteTearDown. */
+int test_suiteTearDown(int num_failures)
+{
+	return generic_suiteTearDown(num_failures);
+}
+
+void setUp(void)
+{
+	mock_lwm2m_client_utils_Init();
+	mock_lwm2m_Init();
+	mock_date_time_Init();
+}
```

Comment:
Missing `mock_lwm2m_resource_ids_Init()` and `mock_lte_lc_Init()`. Also in tearDown the corresponding Verify functions are missing.

Reply:
Is it necessary to have them if no generated `__wrap` function is expected / called in the tests for these APIs? They are only used for types and definitions.

Reply:
Anyways, I added the missing entries.